internal/cmd/get/get_cmd.go

```diff
@@ -197,282 +201,39 @@ func (c *runnerContext) run(cmd *cobra.Command, args []string) error {
+		objects[i], err = c.helper.Get(ctx, id)
 		if err != nil {
-			err = fmt.Errorf("failed to get object with identifier '%s': %w", id, err)
+			return
 		}
-		objects[i] = response.ProtoReflect().Get(c.getObjectFieldDesc).Message().Interface()
 	}
-	result = objects
+	results = objects
 	return
 }
 
-// findObjectTypes finds all the message types that satisfy the conditions to be considered objects that can be handled
-// by this command.
-func (c *runnerContext) findObjectTypes() []protoreflect.MessageDescriptor {
-	var objectDescs []protoreflect.MessageDescriptor
-	protoregistry.GlobalFiles.RangeFiles(func(fileDesc protoreflect.FileDescriptor) bool {
-		serviceDescs := fileDesc.Services()
-		for i := range serviceDescs.Len() {
-			serviceDesc := serviceDescs.Get(i)
-			methodDescs := serviceDesc.Methods()
-
-			// The service must have `List` and `Get` methods:
-			listDesc := methodDescs.ByName(listMethodName)
-			getDesc := methodDescs.ByName(getMethodName)
-			if listDesc == nil || getDesc == nil {
-				continue
-			}
-
-			// The response of the `List` method must have an `items` field, and it must be an list
-			// of objects.
-			listItemsDesc := listDesc.Output().Fields().ByName(itemsFieldName)
-			if listItemsDesc == nil {
-				continue
-			}
-			if listItemsDesc.Cardinality() != protoreflect.Repeated {
-				continue
-			}
-			if listItemsDesc.Kind() != protoreflect.MessageKind {
-				continue
-			}
-
-			// The request of the `Get` method must have an `id` string field:
-			getIdDesc := getDesc.Input().Fields().ByName(idFieldName)
-			if getIdDesc == nil {
-				continue
-			}
-			if getIdDesc.Cardinality() == protoreflect.Repeated {
-				continue
-			}
-			if getIdDesc.Kind() != protoreflect.StringKind {
-				continue
-			}
-
-			// The response of the `Get` method must have an `object` message field:
-			getObjectDesc := getDesc.Output().Fields().ByName(objectFieldName)
-			if getObjectDesc == nil {
-				continue
-			}
-			if getObjectDesc.Cardinality() == protoreflect.Repeated {
-				continue
-			}
-			if getObjectDesc.Kind() != protoreflect.MessageKind {
-				continue
-			}
-
-			// This is a supported type if after the all the above checks it is the type of the
-			// `items` field of the list request and of the `object` field of the get response.
-			if listItemsDesc.Message() != getObjectDesc.Message() {
-				continue
-			}
-			objectDescs = append(objectDescs, listItemsDesc.Message())
-		}
-		return true
-	})
-	return objectDescs
-}
-
-func (c *runnerContext) analyzeDescriptors() error {
-	finders := []func() error{
-		c.findTypeDesc,
-		c.findServiceDesc,
-		c.findListMethodDescs,
-		c.findGetMethodDescs,
+func (c *runnerContext) list(ctx context.Context) (results []proto.Message, err error) {
+	var options reflection.ListOptions
+	if c.filter != "" {
+		options.Filter = c.filter
 	}
-	for _, finder := range finders {
-		err := finder()
-		if err != nil {
-			return err
+	if !c.includeDeleted {
```

Comment:
If this filter is done at the CLI level, would this mean that a GUI would have to implement the same filter if it wants to exclude deleted items? Or is the default behavior at the API level to exclude deleted items?

Reply:
It is done at the CLI level. So yes, the GUI will have to implement the same filter. Note that it can use this exact logic: it only needs to shed `filter=!has(this.metadata.deletion_timestamp)` with its requests.

Reply:
I think we need to keep the deleted objects visible to users because there are lengthy delete operators which they may want to track. For example, I expect that when a cluster is deleted it will take time to actually delete it, and it may even eventually fail. In those situations the way we have to let the user know what is happening is looking at the state of the object.

The limitation right now is that we don't have a mechanism for the server to say "this is completely deleted now". If we had it then we could actually remove the deleted records from the database (move them to an "archive" table) and then this wouldn't be a problem.

My suggestion is to use this filter as temporary solution, till we have the complete solution.

Reply:
Fair enough!